orchagent/routeorch.cpp

```diff
@@ -1672,52 +1714,63 @@ bool RouteOrch::addRoutePost(const RouteBulkContext& ctx, const NextHopGroupKey
         return false;
     }
 
-    /* next_hop_id indicates the next hop id or next hop group id of this route */
-    sai_object_id_t next_hop_id;
-
     if (m_fgNhgOrch->isRouteFineGrained(vrf_id, ipPrefix, nextHops))
     {
         /* Route is pointing to Fine Grained ECMP nexthop group */
         isFineGrained = true;
     }
-    else if (nextHops.getSize() == 0)
-    {
-        /* The route is pointing to a blackhole */
-        blackhole = true;
-    }
-    else if (nextHops.getSize() == 1)
+    /* Check that the next hop group is not owned by NhgOrch. */
+    else if (ctx.nhg_index.empty())
     {
-        /* The route is pointing to a next hop */
-        const NextHopKey& nexthop = *nextHops.getNextHops().begin();
-        if (nexthop.isIntfNextHop())
+        if (nextHops.getSize() == 0)
+        {
+            /* The route is pointing to a blackhole */
+            blackhole = true;
+        }
+        else if (nextHops.getSize() == 1)
         {
-            next_hop_id = m_intfsOrch->getRouterIntfsId(nexthop.alias);
-            /* rif is not created yet */
-            if (next_hop_id == SAI_NULL_OBJECT_ID)
+            /* The route is pointing to a next hop */
+            const NextHopKey& nexthop = *nextHops.getNextHops().begin();
+            if (nexthop.isIntfNextHop())
             {
-                SWSS_LOG_INFO("Failed to get next hop %s for %s",
-                        nextHops.to_string().c_str(), ipPrefix.to_string().c_str());
-                return false;
+                auto next_hop_id = m_intfsOrch->getRouterIntfsId(nexthop.alias);
+                /* rif is not created yet */
+                if (next_hop_id == SAI_NULL_OBJECT_ID)
+                {
+                    SWSS_LOG_WARN("Failed to get next hop %s for %s",
```

Comment:
As mentioned before, while copying existing code, please copy as-is. Please don't change log-levels

Reply:
Done